### PR TITLE
Database model

### PR DESCRIPTION
**Pull request by bensonybs**
model 建立完成，使用者的 API 還需要時間 QQ

### CODE REVIEW

File: migrations/20230323141138-create-account.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.createTable('Accounts', {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER
+      },
+      userId: {
```
Comment (shorty60) [inline migrations/20230323141138-create-account.js:12]: hihi 這裡要設定FK關聯喔
references: {
          model: 'Users',
          key: 'id'
        },

File: migrations/20230323141138-create-account.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      },
+      initialAmount: {
+        allowNull: false,
+        defaultValue: 0.00,
+        type: Sequelize.DECIMAL(10, 2)
+      },
+      description: {
+        type: Sequelize.STRING
+      },
+      accountTypeId: {
```
Comment (shorty60) [inline migrations/20230323141138-create-account.js:28]: 這裡也要設定FK關聯喔
references: {
model: 'AccountTypes',
key: 'id'
},

File: migrations/20230323141202-create-record.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      },
+      userId: {
+        allowNull: false,
+        type: Sequelize.INTEGER
+      },
+      amount: {
+        allowNull: false,
+        type: Sequelize.DECIMAL(10, 2)
+      },
+      categoryId: {
```
Comment (shorty60) [inline migrations/20230323141202-create-record.js:24]: references: {
model: 'RecordCategory',
key: 'id'
},

File: migrations/20230323141202-create-record.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        allowNull: false,
+        type: Sequelize.INTEGER
+      },
+      subCategoryId: {
+        allowNull: false,
+        type: Sequelize.INTEGER
+      },
+      fromAccountId: {
```
Comment (shorty60) [inline migrations/20230323141202-create-record.js:32]: references: {
model: 'Accounts',
key: 'id'
},

File: migrations/20230323141202-create-record.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        type: Sequelize.INTEGER
+      },
+      toAccountId: {
```
Comment (shorty60) [inline migrations/20230323141202-create-record.js:35]: references: {
model: 'Accounts',
key: 'id'
},

File: migrations/20230323141209-create-record-category.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER
+      },
+      name: {
+        allowNull: false,
+        type: Sequelize.STRING
+      },
+      userId: {
+        allowNull: false,
```
Comment (shorty60) [inline migrations/20230323141209-create-record-category.js:17]: references: {
model: 'Users',
key: 'id'
},

File: migrations/20230323141209-create-record-category.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        type: Sequelize.INTEGER
+      },
+      budget: {
+        type: Sequelize.DECIMAL(10, 2)
+      },
+      isDefault: {
```
Comment (shorty60) [inline migrations/20230323141209-create-record-category.js:23]: defaultValue: false,

File: migrations/20230323141218-create-record-sub-category.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER
+      },
+      name: {
+        allowNull: false,
+        type: Sequelize.STRING
+      },
+      userId: {
+        allowNull: false,
```
Comment (shorty60) [inline migrations/20230323141218-create-record-sub-category.js:17]: references: {
model: 'Users',
key: 'id'
},

File: migrations/20230323141218-create-record-sub-category.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+        type: Sequelize.INTEGER
+      },
+      categoryId: {
```
Comment (shorty60) [inline migrations/20230323141218-create-record-sub-category.js:20]: references: {
model: 'RecordCategory',
key: 'id'
},

File: migrations/20230323141218-create-record-sub-category.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+        allowNull: false,
+        type: Sequelize.INTEGER
+      },
+      isDefault: {
```
Comment (shorty60) [inline migrations/20230323141218-create-record-sub-category.js:24]: defaultValue: false,

**Comment by bensonybs** [comment]
新增兩個 commit，再幫我確認一下
1. 增加 model file 的 association 及更新刪除時的連動
2. Default user seed: user1, user2

**Pull request merged**